backend/frontend/views.py

```diff
@@ -295,6 +298,75 @@ def action_start_card(request, card_id):
     )
 
 
+def user_can_request_review(logged_in_user):
+    request = get_current_request()
+    card_id = request.resolver_match.kwargs.get("card_id")
+
+    card = get_object_or_404(AgileCard, pk=card_id)
+    card_assignees = card.assignees.all()
+
+    if logged_in_user in card_assignees:
+        return True
+
+    card_teams = card.get_teams()
+    checker = ObjectPermissionChecker(logged_in_user)
+    checker.prefetch_perms(card_teams)
+
+    return any(
+        (checker.has_perm(Team.PERMISSION_MANAGE_CARDS, team) for team in card_teams)
+    )
+
+
+@csrf_exempt
+@user_passes_test_or_forbidden(user_can_request_review)
+def action_request_review(request, card_id):
+    """The card is in progress or review feedback and the user has chosen to request review"""
+    card = get_object_or_404(AgileCard, id=card_id)
+
+    if len(helpers.agile_card_reviews_outstanding(request.user)):
+        return render(
+            request,
+            "frontend/user/board/js_exec_action_show_card_alert.html",
+            {
+                "card": card,
+                "message": "You have outstanding card reviews.",
```

Comment:
This kind of logic also exists in Ngoako's PR. It wont be DRY. We need functionality to be centralised and referenced, not repeated.